[TSK-978] [TSK-979] ui-kit: ComponentImage, ComponentVideo, ComponentImageCarousel #172

jatinrise opened the pull request:
Component name: `Coursebuilderlesson`

- Added file size tracking to UploadedFileType and related response types.
- Updated FilePreview component to display uploading status with a badge.
- Implemented new Designer and Form components for image files, image galleries, and video files.
- Created stories for image files, image galleries, and video files to showcase new components.
- Refactored course builder types to include new file types and their properties.


Fix #172

jatinrise commented:
Hi @alebg, I've marked this PR as a draft since Manish is currently working on the same Course Builder lesson. Once his PR is merged, I'll sync my changes accordingly. In the meantime, I'd appreciate it if you could review the code.

alebg commented on packages/ui-kit/lib/components/course-builder-lesson-component/image-files.tsx:
```diff
@@ -0,0 +1,102 @@
+                variant="image"
+                file={file}
+                onFilesChange={handleFilesChange}
+                onDelete={handleDelete}
```
@jatinrise I don't understand why onDelete and onDownload are doubled? this looks too confusing as an implementation, but let me know if there's a good reason

alebg commented on packages/ui-kit/lib/components/course-builder-lesson-component/image-files.tsx:
```diff
@@ -0,0 +1,102 @@
+    };
+
+    const handleDelete = () => {
+        setFile(null);
```
@jatinrise I don't think removing the file from the ui-kit is the best. For example, if the delete API call fails, it would seem to the user that it was successful since the file is not there. If this kind of logic is handled by the page instead, then there we can decide better when to visually remove the file

alebg replied:
@jatinrise  @anilrise11 if you're going to go against the guidelines, could you please at least provide arguments for your reasoning? I gave you a clear example of why enforcing this `handleDelete` with no control from the page is problematic.

We've discussed at length several times that the UI kit should be kept "dummy" and most functionalities will be injected as callbacks. Why would we do it differently for this case?

alebg commented on packages/ui-kit/lib/components/course-builder-lesson-component/image-files.tsx:
```diff
@@ -0,0 +1,102 @@
+
+export function DesignerComponent({ elementInstance, locale, onUpClick, onDownClick, onDeleteClick }: DesignerComponentProps) {
+    if (elementInstance.type !== CourseElementType.ImageFile) return null;
+    const dictionary = getDictionary(locale);
```
@jatinrise this `dictionary` is not being used, why is it here?

alebg commented on packages/ui-kit/lib/components/course-builder-lesson-component/image-files.tsx:
```diff
@@ -0,0 +1,102 @@
+ */
+export function FormComponent({ elementInstance, locale }: FormComponentProps) {
+    if (elementInstance.type !== CourseElementType.ImageFile) return null;
+    const dictionary = getDictionary(locale);
```
@jatinrise this is not being used either

alebg reviewed (changes requested):
@jatinrise I don't understand very well, you asked me to review this PR, but a quick check on a single file made me notice that there are many small issues here and there. I know you marked this as a draft, but you asked me to review the code, hence I expect it to be clean enough. I only checked one file but it has unused dictionaries being instantiated, for example. Please clean these details all around the PR, I leave it up to you to check the other files

About the implementation itself, I find it confusing, why are the download and delete functionalities duplicated between the designer layout component and the uploader component?

jatinrise commented:
Hi @alebg, I’ve identified the problem and will address all the issues.

jatinrise commented:
Hi @alebg,
I’ve made the changes as per the following rules:

1. Added callbacks to pass values to the parent
2. Removed unused code
3. Added documentation for better clarity

Please review.

alebg commented:
Thanks @jatinrise . So the approach looks cleaner now. But it still needs some work, right? For example the form stories for the image file don't render:

![immagine](https://github.com/user-attachments/assets/80fd60fd-5525-4e43-bc5a-a1bf40373c5c)

jatinrise commented:
close in favour #199
